cmd/sys-mgmt-executor/main.go

```diff
@@ -148,19 +166,99 @@ func runDockerCommands(service string, operation string) error {
 			expectedStatus = true
 		case STOP:
 			expectedStatus = false
+		case METRICS:
+
 		default:
-			// impossible to get here
 			panic(fmt.Sprintf("invalid operation %s", operation))
 		}
 
 		correctStatus, err := checkDockerContainerStatus(service, expectedStatus)
 		if err != nil {
-			return err
+			return "", err
 		}
 		if !correctStatus {
-			return fmt.Errorf("docker %s operation failed for service %s", operation, service)
+			return "", fmt.Errorf("docker %s operation failed for service %s", operation, service)
+		}
+		return "", nil
+	} else if operation == METRICS {
+
+		LoggingClient.Info(fmt.Sprintf("run the docker stats on service (%s) for operation (%s)", service, operation))
+		cmd := exec.Command("docker", "stats")
```

Comment:
this is a streaming command, which AFAICT will never actually exit. This is not what we should be using to get the output for a single metrics endpoint. Instead I'd recommend using at least `--no-stream` option to get a single snapshot. That would remove the need to do all of the complicated wait group, tickers, timers, and pipes below. Additionally, it would be better to use the `--format` option to get formatted output like what's shown in https://kylewbanks.com/blog/docker-stats-memory-cpu-in-json-format (you don't need to format in json specifically, that's just easy to parse from the stdlib. That also would greatly simplify your parsing code...